### ISSUE TITLE
6.3 Beta - Content width glitch (video proof)

### ISSUE THREAD

**Issue by Ren2049** [WordPress/gutenberg#52657]
### Description

Like you can see in the recording, content width is only applied to the group after removing the value from the content width field for the group. 

In my global styles the content width is set to the same value you see in the video (92vw), like you can see nothing happens when I toggle the button, only after I input and then delete the value the global content width is applied somehow. I know that it's the global value that gets triggered through this action because I've tested it with other values.

### Step-by-step reproduction instructions

1. Install the beta tester plugin and create block theme plugin.
2. Create a blank theme with the create block theme plugin.
3. Set a global content width.
4. Load the index template and apply the global content width to the outer header group and the query loop.
5. Observe that toggling the content width has no effect on the header group but on the query loop.
6. Enter some value into the content width field below the toggle and then delete it using the delete key on your keyboard.
7. Now the content width should be visible.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/133038357/64cfba19-1a2e-4562-96a7-4ce0cb003df5

### Environment info

6.3 Beta 4, no Gutenberg, blank theme created with the official create block plugin.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Ren2049** [comment]
Just noticed that after saving the content width doesn't get applied to the header when I view the site in the browser after it's applied through the editor glitch:

![width2](https://github.com/WordPress/gutenberg/assets/133038357/0b1abfdf-200d-4f4c-a692-a7844c1765ea)

And as you can see the image next to the icon (logotype) isn't aligned in the middle within its row like in the editor view inside video, but bottom aligned (another issue).

As somebody into graphic design this stuff is killing me, I don't even have the nerves to build out more of the site after bumping into these kind of issues. I've opened related issues in the past weeks which either magically solved themselves or got ignored in the end like the alignment bug. 

https://github.com/WordPress/gutenberg/issues/52501#issue-1798585897

https://github.com/WordPress/gutenberg/issues/50720#issue-1714507101

**Comment by carolinan** [comment]
Hi
The header template part in the create block theme "blank theme" includes a group block and a inside the group is a row group block variation. This row is set to full width. That is why the header content is not showing as content width. 
Selecting this block and removing the full width / selecting "none" will reset it to the content width, without needing to change the parent group.

Despite that, there does seem to be a bug here.
I can also see that if I enter a width in to the input field of the parent group, nothing changes until I remove that number.
When I remove the value, the row block inside the group no longer has the full width alignment option in the toolbar:

The "selected" width in the toolbar is still full width.
In the code editor mode, the row still has full width:

```
<!-- wp:group {"layout":{"type":"constrained","contentSize":""}} -->
<div class="wp-block-group"><!-- wp:group {"align":"full","style":{"spacing":{"padding":{"bottom":"30px","top":"30px"}}},"layout":{"type":"flex","justifyContent":"space-between"}} -->
```
The visible width in the editor is no longer full width...
<img width="1512" alt="A header template part with the inner row block selected." src="https://github.com/WordPress/gutenberg/assets/7422055/fd19bacf-725b-48cc-afc4-1ca9bc143afb">

**Comment by Ren2049** [comment]
@carolinan Thanks, setting the row block to none did the trick. Before I didn't look there because I've expected the row block to take the available space. 

I think setting it to none would be a better default setting for the row block of the header template in the blank theme. As I understand full-width is basically a hack to break out of whatever is set on higher level containers. Imo the width should be dictated by the wrapping group.

I'm aware that content-width is mainly used to contain text in blog posts, but why waste it for that if you can use columns on blog post patterns to structure the available space, which is also much better to future proof a design for potential widgets or banners.

I use global content width for all the content of the page, basically to replace horizontal global padding which unfortunately pads everything including the background of patterns.

On blog post templates I typically use 2 columns if I go for sidebars or 3 columns (leaving the outer columns empty or for banners) for a central textblock look, which also works with the global content width. I find that 92vw works really well as a intrinsic content-width setting because it looks good on all sizes leaving enough space towards the screen bezels and doesn't stretch the content to the edges on desktop.
